wicket-native-websocket/wicket-native-websocket-core/src/main/java/org/apache/wicket/protocol/ws/api/message/BinaryMessage.java

```diff
@@ -16,29 +16,57 @@
  */
 package org.apache.wicket.protocol.ws.api.message;
 
+import org.apache.wicket.Application;
+import org.apache.wicket.Session;
+import org.apache.wicket.mock.MockApplication;
+import org.apache.wicket.protocol.ws.api.registry.IKey;
+
 /**
  * A {@link IWebSocketMessage message} with binary data
  *
  * @since 6.0
  */
-public class BinaryMessage implements IWebSocketMessage
+public class BinaryMessage extends AbstractClientMessage
 {
 	private final byte[] data;
 	private final int offset;
 	private final int length;
 
 	/**
-	 * Constructor.
+	 * Not used by Wicket since 8.5.0!
 	 *
 	 * @param data
 	 *      the binary message from the client
 	 * @param offset
 	 *      the offset to read from
 	 * @param length
-	 *      how much data to read
+	 * 		the length of the data to read
 	 */
+	@Deprecated
 	public BinaryMessage(byte[] data, int offset, int length)
 	{
+		this(new MockApplication(), "", new IKey() {}, data, offset, length);
```

Comment:
Is this acceptable ?

It is needed to be able to downport the improvement made in Wicket 9.x.
This constructor is used only by Wicket. I do not expect the user application to have made use of it, but I cannot be fully certain.

Reply:
I see no issues here :)